SonarQube.Client.Tests/SonarQubeService_GetProjectBranchesAsync.cs

```diff
@@ -0,0 +1,113 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarQube.Client.Models;
+
+namespace SonarQube.Client.Tests
+{
+    [TestClass]
+    public class SonarQubeService_GetProjectBranchesAsync : SonarQubeService_TestBase
+    {
+        [TestMethod]
+        public async Task Get_V6_6()
+        {
+            await ConnectToSonarQube("6.6.0.0");
+
+            const string response = @"{
+  ""branches"": [
+    {
+      ""name"": ""feature/foo"",
+      ""isMain"": false,
+      ""type"": ""BRANCH"",
+      ""status"": {
+        ""qualityGateStatus"": ""OK""
+      },
+      ""analysisDate"": ""2017-04-03T13:37:00+0100"",
+      ""excludedFromPurge"": false
+    },
+    {
+      ""name"": ""master"",
+      ""isMain"": true,
+      ""type"": ""BRANCH"",
+      ""status"": {
+        ""qualityGateStatus"": ""ERROR""
```

Comment:
The `type` and `status` fields might be useful in the future, but it isn't documented what values could be, and SonarCloud appears to return different values from SonarQube. So I didn't include them.

Reply:
The `type` looks worrisome, what other values can they return? and should we be filtering based on this?

Reply:
No idea, although looking at real data from SonarCloud and SonarQube, it may be the case that they aren't returning the same values.
However, for our current purposes we're only interested in whether the branch is the main branch or not. I'm making the (possibly unwarranted) assumption that `isMain` will only be true for one branch.